src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -159,7 +159,14 @@ public IChannelHandle GetChannel(CancellationToken cancellationToken)
                     {
                         // ignore exceptions
                     }
-
+                }
+                try
```

Comment:
do we really need this try/catch? As I see we don't cover any other invalidate methods

Reply:
I have the try/catch because we could potentially clear the connection pool, and in the above logic, wrap the try/catch when we clear the connection pool

Reply:
I've removed this extra try/catch as part of the refactor

Reply:
I think in the current version, we protect `Invalidate` logic under try/catch. I would not do it since other Invalidate methods are not protected. I guess it's better to follow to one logic patter in this context

Reply:
I'm fine with this change: it's mildly out of scope but the reasoning seems sound: done!